apps/OboeTester/app/src/main/java/com/mobileer/oboetester/DynamicWorkloadActivity.java

```diff
@@ -385,4 +395,55 @@ public void onStopTest() {
         updateButtons(false);
         super.onStopTest();
     }
+
+
+    @Override
+    public void startTestUsingBundle() {
+        try {
+            StreamConfiguration requestedOutConfig = mAudioOutTester.requestedConfiguration;
+            IntentBasedTestSupport.configureOutputStreamFromBundle(mBundleFromIntent, requestedOutConfig);
+
+            // Specific options.
+            mShouldUseADPF = mBundleFromIntent.getBoolean(KEY_USE_ADPF,
+                    VALUE_DEFAULT_USE_ADPF);
+            mDrawChartAlways =
+                    mBundleFromIntent.getBoolean(KEY_SCROLL_GRAPHICS,
+                            VALUE_DEFAULT_SCROLL_GRAPHICS);
+
+            startTest();
+
+            runOnUiThread(() -> {
+                mPerfHintBox.setChecked(mShouldUseADPF);
```

Comment:
Careful about race conditions here as https://github.com/google/oboe/blob/2de755200046c4821ca5fd220713837a05bb024a/apps/OboeTester/app/src/main/java/com/mobileer/oboetester/DynamicWorkloadActivity.java#L294C22-L294C40 there is a onClickListener for mPerfHintBox.

Reply:
When I ran my tests, the onClickListener did not seem to fire. So I am setting the checkboxes just to make sure the UI is in sync. But maybe it does fire. I will retest this.

Reply:
I put a breakpoint in the onCLickListener and it did not fire when I ran the test using the Intent.
I modified the onClickLister to set mShouldUseADPF.
Does this look OK now?